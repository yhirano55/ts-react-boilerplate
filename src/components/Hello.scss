$color: #888;

.title {
  color: $color;
  font-weight: 700;
}
