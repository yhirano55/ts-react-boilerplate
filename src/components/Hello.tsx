import * as React from "react"
import * as style from "./Hello.scss"

export type HelloProps = {
  compiler: string
  framework: string
}

export const Hello = (props: HelloProps) => (
  <h1 className={style.title}>
    Hello from {props.compiler} and {props.framework}!
  </h1>
)
