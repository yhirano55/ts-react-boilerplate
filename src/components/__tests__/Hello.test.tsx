import * as React from "react"
import * as renderer from "react-test-renderer"
import { shallow } from "enzyme"

import { Hello } from "../Hello"

describe("<Hello />", () => {
  describe("with react-test-renderer", () => {
    const component = renderer.create(
      <Hello compiler="TypeScript" framework="React" />
    )
    it("should display Hello", () => {
      const tree = component.toJSON()
      expect(tree).toMatchSnapshot()
    })
  })

  describe("with enzyme", () => {
    it("should display expected elements", () => {
      const wrapper = shallow(<Hello compiler="TypeScript" framework="React" />)
      expect(wrapper.text()).toBe("Hello from TypeScript and React!")
    })
  })
})
