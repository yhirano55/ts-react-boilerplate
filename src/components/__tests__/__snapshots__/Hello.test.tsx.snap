// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Hello /> with react-test-renderer should display Hello 1`] = `
<h1>
  Hello from 
  TypeScript
   and 
  React
  !
</h1>
`;
